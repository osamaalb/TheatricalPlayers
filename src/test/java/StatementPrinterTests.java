import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.TEXT);

        verify(result);
    }

    @Test
    void statementWithNewPlayTypes() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("henry-v",  new Play("Henry V", "history"));
        plays.put("as-like",  new Play("As You Like It", "pastoral"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        Assertions.assertThrows(Error.class, () -> {
            invoice.print(plays, Invoice.PrintFormat.TEXT);
        });
    }

    @Test
    void testOtherCases() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.TEXT);

        verify(result);
    }

    @Test
    void exampleStatementHTML() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.HTML);

        verify(result);
    }
}
