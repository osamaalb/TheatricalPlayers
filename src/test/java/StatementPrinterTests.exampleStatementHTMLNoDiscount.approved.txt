<!DOCTYPE html>
<html>
<head>
<style>
table, th, td {
border: 1px solid black;
}
th, td {
padding: 5px;
}
</style>
</head>
<body>
<h2>Invoice</h2>
<p><strong>Client:</strong> BigCo</p>
<table>
<tr>
<th>Piece</th>
<th>Seats Sold</th>
<th>Price</th>
</tr>
<tr>
<td>Hamlet</td>
<td>30</td>
<td>$400.00</td>
</tr>
<tr>
<td>As You Like It</td>
<td>20</td>
<td>$360.00</td>
</tr>
<tr>
<td>Othello</td>
<td>40</td>
<td>$500.00</td>
</tr>


<tr>
<td colspan="2" style="text-align: right;"><strong>Total Owed:</strong></td>
<td>$1,260.00</td>
</tr>
<tr>
<td colspan="2" style="text-align: right;"><strong>Fidelity Points Earned:</strong></td>
<td>14</td>
</tr>
<tr>
<td colspan="2" style="text-align: right;"><strong>New Fidelity Points Balance:</strong></td>
<td>149</td>
</tr>
</table>
<p>The invoice is due in 30 days</p>
</body>
</html>